### ISSUE TITLE
Conflicting D symmetry conventions

### ISSUE THREAD

**Issue by pconesa** [asarnow/pyem#33]
Hi, @asarnow thanks for having this nice repo. 
We are actually using it and we were converting cs output to relion star file using this (actually what is in pipy).

Later we did a reconstruction with relion and xmipp and both didn't seem right. Our refinement in CS has a D7 symmetry whereas both reconstructions (relion and xmipp) showed a D14 symmetry (in both cases we specified D7 symmetry fro the reconstruction.

Does it ring any bell?

Alternatively....do CS offer an equivalent to relion_reconstruct, I mean asking CS to make a reconstruction using angle in a star file?

**Comment by pconesa** [comment]
Sorry, we are not using pyem from pypi since it is not there....we are using tag 0.4 from this repo.

**Comment by asarnow** [comment]
Hi,
Right now the release branch is slightly behind. Please try the current master branch.

Unfortunately, there's no way to do a simple reconstruction in cryoSPARC. You might be able to hack this using a reslog job and some restrictive parameters.

About the symmetry, cryoSPARC doesn't restrict the search range to reflect the symmetry. It merely inserts the particles multiple times for each symmetry relation. Try `--sym` with `csparc2star.py` to get all the angles together (or if you used this option, try without, it actually shouldn't affect `relion_reconstruct` with its `--sym` and there could be a bug I didn't find).

**Comment by pconesa** [comment]
@asarnow  thanks for your reply.
What we've seen so far is described here: https://discuss.cryosparc.com/t/dn-symmetry-definition-diferent-as-relion-xmipp-and-maybe-others/3158

There is nothing actually wrong, just that Cryosparc convention for Dn symmetries seems to differ from Relion or Xmipp ones.

We created a script to convert the rodrigues angles to relion angles and compare our output data with the data generated by pyem and both matched. Yo I guess there is no bug here, unless for Dn symmetries you want to convert angles due to differences in the conventions. 

This is our guess, still we would like to hear from cryosparc guys, to confirm or reject our "hypothesis".

Our hypothesis is consistent with what we see reconstructing CS-D7 symmetries in relion or xmipp and getting D14 ones.

**Comment by asarnow** [comment]
Thanks for looking at this. I would like `csparc2star.py` to work out of the box for all symmetries, so I'm reopening the issue for now.

When using symmetry, cryoSPARC only inserts the particles multiple times at each symmetry related pose. It doesn't change the search space as in Relion. The angles you get then go all over the rotation sphere. As such I'm not quite sure why it doesn't work, even if the conventions are different. I'll try and investigate further as well.

**Comment by pconesa** [comment]
@asarnow , we used proteasome data with D7 and made a refinement with cs. After we did a relion reconstruction and a xmipp one both rendering D14 maps. If you want we can provide you with some data. But I guess any D7 reconstruction will render same results.

**Comment by pconesa** [comment]
Although we've most of our test within Scipion, I think we have excluded the possibility that the error we've seen is somewhere in scipion conversions. But, we might have missed something. 

We've compared the star file generated with pyem (cs-->star) and the one used in the relion_reconstruct (actually, due to scipion nature, there is a conversion to .sqlite and then to star file) and both had the same angle information.
